### PR TITLE
octanifies term and vocabulary models.

### PR DESCRIPTION
**Pull request by stopfstedt**

### CODE REVIEW

File: addon/components/dashboard/selected-vocabulary.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import Component from '@glimmer/component';
+import { use } from 'ember-could-get-used-to-this';
+import AsyncProcess from 'ilios-common/classes/async-process';
+
+export default class DashboardSelectedVocabularyComponent extends Component {
+  @use _topLevelTerms = new AsyncProcess(() => [
+    this.args.vocabulary.getTopLevelTerms.bind(this.args.vocabulary),
```
Comment (jrjohnson) [inline addon/components/dashboard/selected-vocabulary.js:7]: Now it gets really interesting 😁

File: addon/components/dashboard/selected-vocabulary.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+  ]);
+
+  get topLevelTerms() {
+    if (!this._topLevelTerms) {
+      return [];
+    }
+    console.log(this._topLevelTerms);
```
Comment (jrjohnson) [inline addon/components/dashboard/selected-vocabulary.js:14]:
```suggestion
```

File: addon/models/term.js [hunk 3]
```diff
@@ -1,171 +1,153 @@
...
+
+  get totalAssociations() {
+    return this.associatedLengths.reduce((prev, curr) => prev + curr);
+  }
+
+  get hasAssociations() {
+    return !!this.totalAssociations;
+  }
+
+  get isTopLevel() {
```
Comment (jrjohnson) [inline addon/models/term.js:70]: This feels too useful to me to get rid of, but I'll accept it.

**Review by jrjohnson — CHANGES_REQUESTED** [review]

File: addon/components/detail-terms-list-item.hbs [hunk 4]
```diff
@@ -11,18 +11,18 @@
       {{@term.description}}
     </IliosTooltip>
   {{/if}}
-  {{#if @term.isTopLevel}}
+  {{#unless (await @term.parent)}}
```
[the remark this answers is not in the document]
Comment (stopfstedt) [reply addon/components/detail-terms-list-item.hbs]: replaced as suggested. :heavy_check_mark:

File: addon/components/detail-terms-list-item.hbs [hunk 5]
```diff
@@ -11,18 +11,18 @@
...
     {{@term.title}}
   {{else}}
-    {{#each (await @term.allParentTitles) as |title|}}
+    {{#each (await this.allParentTitles) as |title|}}
```
[the remark this answers is not in the document]
Comment (stopfstedt) [reply addon/components/detail-terms-list-item.hbs]: :heavy_check_mark:

**Review by jrjohnson — APPROVED** [review]

**Pull request merged**